src/pages/MessageWall.js

```diff
@@ -1,11 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { app, db } from './firebase';
 
 function MessageWall() {
-  return (
-    <div>
-      Message Wall! :D
-    </div>
-  );
-}
+  const [title, setTitle] = useState();
+  const [body, setBody] = useState();
+  const [serviceArea, setServiceArea] = useState();
+  const [mentor, setMentor] = useState();
+  const [caregiver, setCaregiver] = useState();
+  const pinned = false;
+  const myTimestamp = app.firebase.firestore.Timestamp.fromDate(new Date());
+
+  const data = {
+    title,
+    body,
+    serviceArea: [serviceArea],
+    target: ['mentor', 'caregiver'],
```

Comment:
For this part, I think you should change target depending on if mentor and caregiver are checked or not- right now, it sets mentor and caregiver no matter how the checkboxes are checked